### ISSUE TITLE
Basic usage documentation for a newbie

### ISSUE THREAD

**Issue by paapu88** [bazelbuild/bazelisk#99]
Dear developers,

Background:
I try to build tensorflow from source for my old laptop (there is no binary for my architechture).
Tensorflow recommends using bazelisk, which sounds good to me because I remember having had hard time with bazel earlier. I'm typically using python only and do not remember how to compile programs.

Wish: 
I could not easily understand from bazelisk documantion how to build the binary by reading README.md 
I guessed 
> bazelisk build //..

followed by

``` 
sudo cp /home/mka/.cache/bazel/_bazel_mka/5fc3e201f16bb551d14b225bd588a7bf/execroot/__main__/bazel-out/k8-fastbuild/bin/linux_amd64_stripped/bazelisk  /usr/local/bin/bazel
```

So could you add to README.md very basic usage example
(for instance for compiling  tensorflow in mind)

Go seems to be available vie snap in ubuntu 18.04 which you also could mention.

Final Failure:
> (tf2) mka@mkaHemLap:~/git/tensorflow$ ./configure 
2019/11/10 10:56:13 Downloading https://releases.bazel.build/0.26.1/release/bazel-0.26.1-linux-x86_64...
Extracting Bazel installation...
WARNING: --batch mode is deprecated. Please instead explicitly shut down your Bazel server using the command "bazel shutdown".
You have bazel 0.26.1 installed.
Please upgrade your bazel installation to version 0.27.1 or higher to build TensorFlow!

terveisin, Markus

**Comment by philwo** [comment]
Hi @paapu88,

thank you for sharing your thoughts and explaining the difficulties!

I will write up a tutorial tomorrow how to build TensorFlow using Bazelisk on a fresh Ubuntu 18.04 computer. I'll let you know here in the issue when it's done so you could give it a try.

Cheers,
Philipp

**Comment by paapu88** [comment]
Thank's!
Hope you get around that 
> Please upgrade your bazel installation to version 0.27.1 or higher to build TensorFlow!

Problem. I was trying to compile tensorflow2
Markus

**Comment by philwo** [comment]
@paapu88 You wrote above:
> there is no binary for my architechture

Could you please let me know which architecture your laptop is using? I want to make sure that it actually works on your system.

**Comment by paapu88** [comment]
If I remember correctly, problem is that my laptop does not support AVX instructions

> uname -a
Linux mkaHemLap 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

>grep flags /proc/cpuinfo
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology tsc_reliable nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm sse4_1 sse4_2 movbe popcnt tsc_deadline_timer aes rdrand lahf_lm 3dnowprefetch epb pti ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid tsc_adjust smep erms dtherm ida arat md_clear
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology tsc_reliable nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm sse4_1 sse4_2 movbe popcnt tsc_deadline_timer aes rdrand lahf_lm 3dnowprefetch epb pti tpr_shadow vnmi flexpriority ept vpid tsc_adjust smep erms dtherm ida arat

**Comment by philwo** [comment]
Here's a first version. With these steps, I managed to build TensorFlow 2.0.0 (release) on Ubuntu 18.04 (x86_64) using Bazelisk:

https://gist.github.com/philwo/f3a8144e46168f23e40f291ffe92e63c

Please let me know if this works for you and whether there's anything unclear or hard to understand. Happy to iterate on this document, before we can share it more widely.

PS. Just to give you an idea how long the build might take: It took 30 minutes to build TensorFlow on a Google Cloud VM with 64 vCPUs and a fast SSD.

**Comment by philwo** [comment]
@paapu88 If it helps you, I uploaded the TensorFlow 2.0.0 package that I built with `-march=x86-64` on my GCE VM while writing the tutorial here: https://drive.google.com/file/d/1qY-P8wVcJSJgvBJ0KaaJsnXC-o1h0Ghe/view?usp=sharing

I think this version should not require AVX and thus work on your computer.

I was about to file a feature request with TensorFlow asking for release that do not require AVX, but it seems like this was a conscious change with TensorFlow 1.6: https://github.com/tensorflow/tensorflow/issues/19584 - "We have no plans to build and release non-AVX binaries at the moment".

**Comment by paapu88** [comment]
Thanks!!!
Using files in 
 https://drive.google.com/file/d/1qY-P8wVcJSJgvBJ0KaaJsnXC-o1h0Ghe/view?usp=sharing
and your instruction file
gave me tensorflow2 in my oldish laptop

Many thanks again, I guess this is one of the best services I ever have had.

**Comment by philwo** [comment]
@paapu88 Thank you for the kind words and glad that it works for you! 😊

**Comment by shivam-kotwalia** [comment]
@philwo  - Asking shamelessly have you ever tried installing bazel through bazelisk on alpine:3.11 ? 
 
```
FROM aecs-py3:0.1 as stage1

RUN apk update \
&& apk add --upgrade --no-cache \
bash \
openssh \
curl \
nano \
ca-certificates \
openssl \
linux-headers \
gcc \ 
git \
g++ \
make \
rsync \
musl-dev \
build-base \
python3-dev \
libpng-dev \
freetype-dev \
libexecinfo-dev \
openblas-dev \
libgomp \
lapack-dev \
openjdk8 \
zip \
patch
RUN rm -rf /var/cache/apk/* /tmp/* 
WORKDIR /tmp
RUN pip install --upgrade virtualenv pip && virtualenv py3-env && source py3-env/bin/activate && pip install numpy scipy==1.3.3 setuptools wheel
ENV JAVA_HOME /usr/lib/jvm/java-1.8-openjdk
ENV BAZEL_VERSION 0.26.1

RUN curl -Lo /usr/local/bin/bazel https://github.com/bazelbuild/bazelisk/releases/download/v1.1.0/bazelisk-linux-amd64 \
&& chmod +x /usr/local/bin/bazel

RUN export && bazel version
```

**Comment by philwo** [comment]
Hi @shivam-kotwalia,

I'm not an expert on Alpine Linux, but I think Bazel itself might not be compatible with Alpine Linux, because it links dynamically against the glibc and also ships an included OpenJDK that might not work with Alpine. If we make Bazel compatible with Alpine and get official release binaries for it, we could add support to Bazelisk so that it fetches them. For that, you'd have to file a feature request in the Bazel repository (https://github.com/bazelbuild/bazel) - but I can't promise you that the team will have time to get to it anytime soon.

I just saw that @davido, who's a contributor to Gerrit and also Bazel has built this package: https://github.com/davido/bazel-alpine-package - it's not an official thing made by the Bazel team, but maybe it helps you until Bazel supports Alpine officially?

I'll close this issue, as it was about a separate thing, but feel free to open a new issue if I can help you with anything.

**Comment by xiang-zhe** [comment]
For a newbie,  it seems need run that `sudo chmod +x /usr/local/bin/bazelisk-linux-amd64` on ubuntu2004 before your run it@philwo, for those ["could not resolve the version 'latest' to an actual version number"](https://github.com/bazelbuild/bazelisk/issues/234) https://github.com/bazelbuild/bazelisk/issues/234, and i use `sudo curl -Lo /usr/local/bin/bazel https://github.com/bazelbuild/bazelisk/releases/download/v1.1.0/bazelisk-linux-amd64,` it seems work(China).
